numba_rvsdg/tests/test_figures.py

```diff
@@ -0,0 +1,479 @@
+        type: python_bytecode
+        begin: 8
+        end: 12
+    python_bytecode_block_3:
+        type: python_bytecode
+        begin: 12
+        end: 14
+    python_bytecode_block_4:
+        type: python_bytecode
+        begin: 14
+        end: 18
+    python_bytecode_block_5:
+        type: python_bytecode
+        begin: 18
+        end: 20
+    synth_asign_block_0:
+        type: synth_asign
+        variable_assignment: {'control_var_0': 0}
+    synth_asign_block_1:
+        type: synth_asign
+        variable_assignment: {'control_var_0': 1}
+    synth_asign_block_2:
+        type: synth_asign
+        variable_assignment: {'control_var_0': 2}
+    synth_fill_block_0:
+        type: synth_fill
+    synth_head_block_0:
+        type: synth_head
+        branch_value_table: {0: 'branch_region_4', 2: 'branch_region_4', 1: 'branch_region_5'} # noqa
+        variable: control_var_0
+    synth_tail_block_0:
+        type: synth_tail
+    tail_region_0:
+        type: region
+        kind: tail
+        contains: ['branch_region_4', 'branch_region_5', 'head_region_2', 'tail_region_2'] # noqa
+        header: head_region_2
+        exiting: tail_region_2
+        parent_region: meta_region_0
+    tail_region_1:
+        type: region
+        kind: tail
+        contains: ['synth_tail_block_0']
+        header: synth_tail_block_0
+        exiting: synth_tail_block_0
+        parent_region: branch_region_0
+    tail_region_2:
+        type: region
+        kind: tail
+        contains: ['python_bytecode_block_5']
+        header: python_bytecode_block_5
+        exiting: python_bytecode_block_5
+        parent_region: tail_region_0
+edges:
+    branch_region_0: ['tail_region_0']
+    branch_region_1: ['tail_region_0']
+    branch_region_2: ['tail_region_1']
+    branch_region_3: ['tail_region_1']
+    branch_region_4: ['tail_region_2']
+    branch_region_5: ['tail_region_2']
+    head_region_0: ['branch_region_0', 'branch_region_1']
+    head_region_1: ['branch_region_2', 'branch_region_3']
+    head_region_2: ['branch_region_4', 'branch_region_5']
+    python_bytecode_block_0: ['branch_region_0', 'branch_region_1']
+    python_bytecode_block_1: ['branch_region_2', 'branch_region_3']
+    python_bytecode_block_2: ['synth_asign_block_1']
+    python_bytecode_block_3: ['synth_asign_block_2']
+    python_bytecode_block_4: ['tail_region_2']
+    python_bytecode_block_5: []
+    synth_asign_block_0: ['tail_region_0']
+    synth_asign_block_1: ['tail_region_1']
+    synth_asign_block_2: ['tail_region_1']
+    synth_fill_block_0: ['tail_region_2']
+    synth_head_block_0: ['branch_region_4', 'branch_region_5']
+    synth_tail_block_0: ['tail_region_0']
+    tail_region_0: []
+    tail_region_1: ['tail_region_0']
+    tail_region_2: []
+backedges:"""
+
+
+class TestBahmannFigures(SCFGComparator):
+    def test_figure_3(self):
+        # Figure 3 of the paper
+
+        # fake bytecode just good enough for FlowInfo
+        bc = [
```

Comment:
Can this be refactored to be input YAML? IIRC this style of `bc` dates from a time before any kind of YAML input had existed.

Reply:
The bytecode is still being stored separately in `bcmap` and not in SCFG. I think this refactor can be made once #75 is done and bytecode is a part of SCFG.

Reply:
OK, I have created an issue as a reminder: https://github.com/numba/numba-rvsdg/issues/85